src/util/address.rs

```diff
@@ -491,6 +507,7 @@ impl Address {
                         32 => Some(AddressType::P2wsh),
                         _ => None,
                     },
+                    WitnessVersion::V1 => Some(AddressType::P2tr),
```

Comment:
Yes, but the length is checked during the struct init. We do length check above only because we need to distinguish two addresses

Reply:
No, we do the check above because Segwit requires program len to be of size 32 or 20, otherwise we must error. BIP341 specifically mentions 
```
 The following rules only apply when such an output is being spent. Any other outputs, including version 1 outputs with lengths other than 32 bytes, or P2SH-wrapped version 1 outputs[3], remain unencumbered.
```
So, presently this is a bug as this will mark other programs as taproot when they are not.